### PR TITLE
Parallelize flexibledecay2

### PR DESCRIPTION
**Pull request by wkotlarski**
This parallelizes calculation of decay amplitudes at the mathematica level. There were two possible ways to do it:

1. parallelize calculation of decays of a single particle
2. calculate decays of multiple particles at once

I went with option 2. I don't know if option 1 would be slower or faster.

Example times it takes to calculate decay amplitudes (I'm timing only the part I'm parallelizing) on my laptop:

| model        | sequential           | parallel  |
| ------------- |:-------------:| -----:|
| MSSM     | 190s | 140s |
| MRSSM2      | 555s     |   395s |

### CODE REVIEW

File: meta/Vertices.m [hunk 1]
```diff
@@ -61,13 +61,32 @@
 
 Begin["`Private`"]
 
+FSVertex[sortedFields_, UseDependences_:False] := FSVertex[sortedFields, UseDependences] =
+Module[{vertexList, vertex},
+
+   vertexList =
+      Switch[Length[sortedFields],
+		   3, SARAH`VertexList3,
```
Comment (Expander) [inline meta/Vertices.m:69]: Please check the indentation of these lines. :)

File: CHANGES.rst [hunk 2]
```diff
@@ -4,9 +4,28 @@ FlexibleSUSY 2.6.2 [July, 08 2021]
...
+  Example speedup:
+  .. list-table::
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - model
+     - sequential
+     - parallel
+   * - `SM`
+     -
```
Comment (Expander) [inline CHANGES.rst:17]: Will this table be filled before the merge?

**Comment by Expander** [comment]
Looks good to me! Are you planning to create a paper for these new developments?

File: meta/Vertices.m (hunk 1 above)
Comment (wkotlarski) [reply meta/Vertices.m:69]: Sorry Alex, but I don't see it. Is there a problem? It's consistently indented with 3 spaces.

File: CHANGES.rst (hunk 2 above)
Comment (wkotlarski) [reply CHANGES.rst:17]: Yes. Sorry. I forgot about it. I'll copy a table from this PR.

**Comment by wkotlarski** [comment]
I didn't plan to write any paper. In the end it's mostly technical stuff. Nice but not publishable.

File: meta/Vertices.m (hunk 1 above)
Comment (Expander) [reply meta/Vertices.m:69]: I can see it in the github diff of changes. I think it is due to a mixture of tabs and spaces: line 69 has 2 tabs and 4 spaces. Line 70 has 3 tabs. (Sorry, but it is actually just a minor and pedantic thing. :) )

File: CHANGES.rst (hunk 2 above)
Comment (Expander) [reply CHANGES.rst]: Thanks!

**Comment by Expander** [comment]
> I didn't plan to write any paper. In the end it's mostly technical stuff. Nice but not publishable.

Ok, I was just curious because I think I saw an arxiv preprint number somewhere. :)

File: meta/Vertices.m (hunk 1 above)
Comment (wkotlarski) [reply meta/Vertices.m:69]: You're right. My bad. It's fixed.

Comment (Expander) [reply meta/Vertices.m]: Thanks a lot!

**Review by Expander — APPROVED** [review]

**Comment by wkotlarski** [comment]
@uukhas, do you have any comments?

File: meta/FlexibleSUSY.m [hunk 3]
```diff
@@ -1986,18 +1986,65 @@ corresponding tadpole is real or imaginary (only in models with CP
             partialWidthCalculationPrototypes = "", partialWidthCalculationFunctions = "",
             calcAmplitudeSpecializationDecls = "", calcAmplitudeSpecializationDefs = "",
             partialWidthSpecializationDecls = "", partialWidthSpecializationDefs = "",
-            smParticleAliases, solverIncludes = "", solver = ""},
+            smParticleAliases, solverIncludes = "", solver = "", contentOfPath = $Path, modelName},
+
+            modelName =
+               If[SARAH`submodeldir =!= False,
+                  SARAH`modelDir <> "/" <> SARAH`submodeldir,
```
Comment (uukhas) [inline meta/FlexibleSUSY.m:1993]: If it is a `modelName`, then shouldn't it be via `-`, like `Leptoquarks-ScalarLeptoquarks.m`?

File: meta/Utils.m [hunk 4]
```diff
@@ -410,7 +410,7 @@ occurrence of the given rule is replaced (if it exists) or added (if
       Quit[1];
    ];
 ];
-SetAttributes[{AssertOrQuit,internalAssertOrQuit},{HoldAll,Locked,Protected}];
```
Comment (uukhas) [inline meta/Utils.m:413]: Does `Protected` break something?

File: meta/Utils.m [hunk 5]
```diff
@@ -461,7 +461,7 @@ occurrence of the given rule is replaced (if it exists) or added (if
    );
    Internal`HandlerBlock[{"MessageTextFilter", Filter}, expression]
 ];
-SetAttributes[{EvaluateOrQuit,internalEvaluateOrQuit},{HoldAll,Locked,Protected}];
```
Comment (uukhas) [inline meta/Utils.m:464]: Does `Protected` break something?

**Review by uukhas — APPROVED** [review]

**Pull request merged**